### DISCUSSION TITLE
Extending layer in private GitHub organization repository

### DISCUSSION THREAD

**Discussion by maxdzin** [nuxt/nuxt#23293] [Questions] [answered]
Good day everyone!

Can you point me to how to extend the **GitHub private organization** layer app repo?

The other Nuxt app repositories those about to extend from that layer app repository are within the same organization.

In docs, it's just mentioned that the `GIGET_AUTH` token needs to be specified:
https://nuxt.com/docs/guide/going-further/layers#git-repository
```
If you want to extend a private remote source, you need to add the environment variable GIGET_AUTH=<token> to provide a token.
```

However, I have not yet found how to generate one for the organization.

In GitHub docs I just found out about how to set the policy, but nothing about generating the token:
https://docs.github.com/en/organizations/managing-programmatic-access-to-your-organization/setting-a-personal-access-token-policy-for-your-organization

**Comment by KeJunMao** [comment]
You can view the implementation details here https://github.com/unjs/giget/commit/88533428f239da561f1ada31b68127c746d1837a#diff-2b7ff992842c6f8113f826af7e81b310a87148e85eef65304868dfc91dad5feb

This means that just create a [personal access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens)

**Comment by madsh93** [comment]
Be aware that layers don't work on Vercel and Netlify.

https://github.com/nuxt/nuxt/issues/22234

**Comment by maxdzin** [comment, marked as answer]
OK. As I figured out, there is currently no way to create a private access token for an organization.
As a workaround, a "special" administrator account was created for the GitHub organization, and a private access token was set for it. Then it was used in the repositories exclusively of this organization.

**Reply by ruzlicali-dev** [reply]
Do you have step by step on how to do this. 

I just want to extends a Private Repo under a Organization and my Account was also `Owner` on that Organization.

How to extends the Private Repos into my Nuxt Layer Projects.

Thank you.

**Reply by maxdzin** [reply]
Hello @ruslicali 
I tried it a long time ago (~6 months ago)... maybe that hasn't changed since then too much.

In the Nuxt project that will use the layer, it is need to specify the layer as this:
```ts
export default defineNuxtConfig({
  extends: [
    ['github:ORGANIZATION/REPOSITORY', { auth: process.env.GITHUB_TOKEN, install: true }]
  ]
})
```

Then you will need to get the personal access token for your repo, as I remember, you can define it in the repository settings:
https://github.com/organizations/YOUR_ORGANIZATION/settings/personal-access-tokens-onboarding

and use it in your project .env:
```.env
GITHUB_TOKEN=YOUR_PRIVATE_TOKEN
```

I hope that help

**Reply by BaptisteCrouzet** [reply]
See : https://github.com/unjs/giget?tab=readme-ov-file#providing-token-for-private-repositories

And this :

> Note: For github private repository access with Fine-grained access tokens, you need to give Contents and Metadata repository permissions.

**Comment by maxdzin** [comment]
@madsh93 Yep. It is so. Unfortunately, so far the use of layers is very limited.
It's a cool idea, but it still doesn't work for some scenarios (and especially some large projects).
So, for now, we will have to wait (and maybe contribute if possible) until that will be implemented sufficiently.

**Comment by blasdfaa** [comment]
I also encountered a limitation on the use of layers for private repositories of an organization. I would like to have this opportunity
